include/kernel.h

```diff
@@ -4102,6 +4102,42 @@ static inline char *K_THREAD_STACK_BUFFER(k_thread_stack_t sym)
 
 #endif /* _ARCH_DECLARE_STACK */
 
+
+#ifdef CONFIG_PRINTK
+/**
+ * @brief Emit one character to the console device
+ *
+ * This is used internally by printk() and related functions.
+ *
+ * @param c Character value to print
+ * @return The printed character, or 0 if nothing was printed
+ */
+extern int _k_char_out(int c);
```

Comment:
Seems like "internal" stuff wants to go somewhere else, no?  kernel.h is the core API header.  Stuff for the library code might be better placed in a "klib.h" or "ikernel.h" or something.  Or split it out and call it "kernel/console.h" or whatever.

Reply:
Keep in mind that these three functions are actually part of the kernel, they are implemented in kernel/printk.c and *not* in the runtime library. They are put in the kernel directly because they manage the global routing of incoming characters to console drivers.

Reply:
I moved this variously to zephyr_runtime.h and nano_internal.h